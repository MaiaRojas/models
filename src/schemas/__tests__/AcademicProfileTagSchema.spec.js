const mongoose = require('mongoose/browser');
const AcademicProfileTagSchema = require('../AcademicProfileTagSchema')(mongoose);
const TagSchema = require('../TagSchema')(mongoose);
const UserSchema = require('../UserSchema')(mongoose);

describe('AcademicProfileTagSchema', () => {
  it('should fail validation when missing fields are provided', () => {
    const doc = new mongoose.Document({}, AcademicProfileTagSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should successfully validate with proper values', (done) => {
    const createdBy = new mongoose.Document({
      uid: 'xxxxxxxxxx',
      email: 'foo@bar.baz',
    }, UserSchema);
    const tag = new mongoose.Document({}, TagSchema);

    const doc = new mongoose.Document({
      createdBy: createdBy._id,
      tag: tag._id,
      assignmentReason: 'foo',
    }, AcademicProfileTagSchema);

    return doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
