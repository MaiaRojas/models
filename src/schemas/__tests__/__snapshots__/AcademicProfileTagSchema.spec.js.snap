// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcademicProfileTagSchema should fail validation when missing fields are provided 1`] = `
Object {
  "assignmentReason": [ValidatorError: Path \`assignmentReason\` is required.],
  "createdBy": [ValidatorError: Path \`createdBy\` is required.],
  "tag": [ValidatorError: Path \`tag\` is required.],
}
`;
